src/Generators/ControllerTagGenerator.php

```diff
@@ -39,6 +42,26 @@ protected function generateControllerObjectTags()
             if ($pageClassname !== 'Page') {
                 $this->pushMixinTag($pageClassname);
             }
+        } else if ($this->isContentController($this->className)) {
+            if (empty(self::$pageClassesCache)) {
+                self::$pageClassesCache = ClassInfo::subclassesFor(Page::class);
+            }
+
+            foreach (self::$pageClassesCache as $pageClassname) {
+                if (Config::inst()->get($pageClassname, 'controller_name') == $this->className) {
+                    $pageClassname = $this->getAnnotationClassName($pageClassname);
+
+                    $this->pushPropertyTag($pageClassname . ' dataRecord');
+                    $this->pushMethodTag('data()', $pageClassname . ' data()');
```

Comment:
As I said below, can you rewrite this to a `sprintf('%s data()')`? (and same for the dataRecord)